library/core/src/intrinsics.rs

```diff
@@ -2095,6 +2097,65 @@ extern "rust-intrinsic" {
     /// `ptr` must point to a vtable.
     /// The intrinsic will return the alignment stored in that vtable.
     pub fn vtable_align(ptr: *const ()) -> usize;
+
+    /// Selects which function to call depending on the context.
+    ///
+    /// If this function is evaluated at compile-time, then a call to this
+    /// intrinsic will be replaced with a call to `called_in_const`. It gets
+    /// replaced with a call to `called_at_rt` otherwise.
+    ///
+    /// # Type Requirements
+    ///
+    /// The two functions must be both function items. They cannot be function
+    /// pointers or closures.
+    ///
+    /// `arg` will be the arguments that will be passed to either one of the
+    /// two functions, therefore, both functions must accept the same type of
+    /// arguments. Both functions must return RET.
```

Comment:
`arg` must be a tuple, right?

Reply:
Yes, but that is enforced by the function signature where `F: FnOnce<ARG>`.

Reply:
docs updated.